File: src/test/java/org/railwaystations/rsapi/adapter/in/web/controller/MockMvcTestConfiguration.java
package org.railwaystations.rsapi.adapter.in.web.controller;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.railwaystations.rsapi.adapter.out.mastodon.MastodonBotConfig;
import org.railwaystations.rsapi.adapter.out.mastodon.MastodonBotHttpClient;
import org.railwaystations.rsapi.adapter.out.monitoring.MockMonitor;
import org.railwaystations.rsapi.adapter.out.photostorage.WorkDir;
import org.railwaystations.rsapi.core.ports.out.MastodonBot;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Profile;

import java.io.IOException;
import java.nio.file.Files;

@TestConfiguration
@Profile({ "mockMvcTest" })
public class MockMvcTestConfiguration {

    @Bean
    MastodonBot mastodonBot(final ObjectMapper objectMapper) {
        return new MastodonBotHttpClient(new MastodonBotConfig(null, null, null), objectMapper);
    }

    @Bean
    WorkDir workDir() throws IOException {
        return new WorkDir(Files.createTempDirectory("rsapi").toString(), null);
    }

    @Bean
    MockMonitor monitor() {
        return new MockMonitor();
    }

}

File: src/test/kotlin/org/railwaystations/rsapi/adapter/web/controller/MockMvcTestConfiguration.kt
package org.railwaystations.rsapi.adapter.web.controller

import com.fasterxml.jackson.databind.ObjectMapper
import org.railwaystations.rsapi.adapter.mastodon.MastodonBotConfig
import org.railwaystations.rsapi.adapter.mastodon.MastodonBotHttpClient
import org.railwaystations.rsapi.adapter.monitoring.FakeMonitor
import org.railwaystations.rsapi.adapter.photostorage.WorkDir
import org.railwaystations.rsapi.core.ports.MastodonBot
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Profile
import java.nio.file.Files

@TestConfiguration
@Profile("mockMvcTest")
class MockMvcTestConfiguration {
    @Bean
    fun mastodonBot(objectMapper: ObjectMapper): MastodonBot {
        return MastodonBotHttpClient(MastodonBotConfig("", "", ""), objectMapper)
    }

    @Bean
    fun workDir(): WorkDir {
        return WorkDir(Files.createTempDirectory("rsapi").toString(), null)
    }

    @Bean
    fun monitor(): FakeMonitor {
        return FakeMonitor()
    }
}
